mistral/mistral-7b-instruct-chat-trt-llm/packages/client.py

```diff
@@ -122,34 +56,106 @@ def load_server_and_model(self, env: dict):
-            return False
+        command = self.build_server_start_command(mpi, env)
+        return subprocess.Popen(
+            command,
+            env={**os.environ, **env},
+        )
+
+    async def infer(
+        self,
+        request_id: str,
+        prompt: str,
+        max_tokens: int = 50,
+        beam_width: int = 1,
+        bad_words: list = [""],
+        stop_words: list = [""],
+        stream: bool = True,
+        repetition_penalty: float = 1.0,
+        ignore_eos: bool = False,
+        eos_token_id: int = None,
+        model_name="ensemble",
+    ):
+        """Infer a response from the model."""
+        if not self.grpc_client_instance:
+            self.start_grpc_stream()
+
+        prompt_data = np.array([[prompt]], dtype=object)
+        output_len_data = np.ones_like(prompt_data).astype(np.uint32) * max_tokens
+        bad_words_data = np.array([bad_words], dtype=object)
+        stop_words_data = np.array([stop_words], dtype=object)
+        stream_data = np.array([[stream]], dtype=bool)
+        beam_width_data = np.array([[beam_width]], dtype=np.uint32)
+        repetition_penalty_data = np.array([[repetition_penalty]], dtype=np.float32)
+
+        inputs = [
+            prepare_grpc_tensor("text_input", prompt_data),
+            prepare_grpc_tensor("max_tokens", output_len_data),
+            prepare_grpc_tensor("bad_words", bad_words_data),
+            prepare_grpc_tensor("stop_words", stop_words_data),
+            prepare_grpc_tensor("stream", stream_data),
+            prepare_grpc_tensor("beam_width", beam_width_data),
+            prepare_grpc_tensor("repetition_penalty", repetition_penalty_data),
+        ]
 
-        result = None
+        if not ignore_eos:
+            assert (
+                eos_token_id is not None
+            ), "eos_token_id must be provided if ignore_eos is False"
+            end_id_data = np.array([[eos_token_id]], dtype=np.uint32)
+            inputs.append(prepare_grpc_tensor("end_id", end_id_data))
 
-        while not _is_final_response(result):
-            try:
-                result = user_data._completed_requests.get()
-                if not isinstance(result, InferenceServerException):
-                    res = result.as_numpy("text_output")
-                    yield res[0].decode("utf-8")
-                else:
-                    yield json.dumps({"status": "error", "message": result.message()})
-            except Exception as e:
-                yield json.dumps({"status": "error", "message": str(e)})
-                break
+        async def input_generator():
+            yield {"model_name": model_name, "inputs": inputs, "request_id": request_id}
+
+        result_iterator = self.grpc_client_instance.stream_infer(
+            inputs_iterator=input_generator(),
+        )
+
+        async for result in result_iterator:
+            if not isinstance(result, InferenceServerException):
+                res = result[0].as_numpy("text_output")
```

Comment:
@aspctu https://github.com/triton-inference-server/client/blob/main/src/python/examples/simple_grpc_aio_sequence_stream_infer_client.py#L140 unpack tuple to check for error state